src/SamplesApp/SamplesApp.UITests/Windows_UI_Xaml_Controls/ToggleSwitchTests/UnoSamples_Tests.ToggleSwitch.cs

```diff
@@ -37,5 +37,28 @@ public void ToggleSwitch_TemplateReuseTest()
 			Assert.AreEqual("False", toggleSwitchGroup.GetDependencyPropertyValue("IsOn")?.ToString());
 			Assert.AreEqual("True", separatedToggleSwitch.GetDependencyPropertyValue("IsOn")?.ToString());
 		}
+
+		[Test]
+		[AutoRetry]
+		public void ToggleSwitch_HeaderTest()
+		{
+			Run("UITests.Shared.Windows_UI_Xaml_Controls.ToggleSwitchControl.ToggleSwitch_Header");
+
+			var toggleSwitchWithHeader = _app.Marked("toggleSwitchWithHeader");
+			_app.WaitForElement(toggleSwitchWithHeader);
+
+			var toggleSwitchHeader = toggleSwitchWithHeader.Marked("toggleSwitchWithHeader");
+			_app.WaitForElement(toggleSwitchHeader);
+
+			var toggleSwitchHeaderContentTextBlock = _app.Marked("toggleSwitchHeaderContent");
+			_app.WaitForElement(toggleSwitchHeaderContentTextBlock);
+
+			var toggleSwitchNoHeader = _app.Marked("toggleSwitchNoHeader");
+			_app.WaitForElement(toggleSwitchNoHeader);
+
+			Assert.AreEqual("Visible", toggleSwitchWithHeader.GetDependencyPropertyValue("Visibility").ToString());
+			Assert.AreEqual("Test ToggleSwitch Header", toggleSwitchHeaderContentTextBlock.GetDependencyPropertyValue("Text").ToString());
+			Assert.IsNull(toggleSwitchNoHeader.GetDependencyPropertyValue("Header.Content"));
```

Comment:
```suggestion
			Assert.AreEqual("", toggleSwitchNoHeader.GetDependencyPropertyValue("Header"));
```

When testing for this property, the UI Testing API cannot provide "null", but only a string representation of the property. Also, the `GetDependencyPropertyValue` only takes one property of the owner, and does not behave lie a binding path. Testing for `Header` here is enough.